run-build-functions.sh

```diff
@@ -614,13 +631,30 @@ install_dependencies() {
 
     restore_home_cache ".node/corepack" "corepack dependencies"
 
-    if [ "$NETLIFY_USE_YARN" = "true" ] || ([ "$NETLIFY_USE_YARN" != "false" ] && [ -f yarn.lock ]); then
-      run_yarn $YARN_VERSION "$featureFlags"
-    elif [ "$NETLIFY_USE_PNPM" = "true" ] || ([ "$NETLIFY_USE_PNPM" != "false" ] && [ -f pnpm-lock.yaml ]); then
-      run_pnpm $PNPM_VERSION "$featureFlags"
-    else
-      run_npm "$featureFlags"
-    fi
+		if has_feature_flag "$featureFlags" "build-image_use_new_package_manager_detection"; then
+			local pkgManager=$(echo "$buildInfo" | jq -r '.packageManager.name')
+			case $pkgManager in
+				"yarn")
+					run_yarn "$YARN_VERSION" "$featureFlags"
+					;;
+				"pnpm")
+					run_pnpm "$PNPM_VERSION" "$featureFlags"
+					;;
+				*)
+					run_npm "$featureFlags"
+					;;
+			esac
+
+		else
+			# feature flag turned off use the old detection
+      if [ "$NETLIFY_USE_YARN" = "true" ] || ([ "$NETLIFY_USE_YARN" != "false" ] && [ -f yarn.lock ]); then
+        run_yarn $YARN_VERSION "$featureFlags"
+      elif [ "$NETLIFY_USE_PNPM" = "true" ] || ([ "$NETLIFY_USE_PNPM" != "false" ] && [ -f pnpm-lock.yaml ]); then
+        run_pnpm $PNPM_VERSION "$featureFlags"
+      else
+        run_npm "$featureFlags"
+      fi
+		fi # end ifelse feature flag
```

Comment:
Indentation looks off?

Reply:
You are such a ninja how did you spot the wrong tab indent instead of the spaces? 🤔  for me it looks really the same 😂